### DISCUSSION TITLE
Precedence of `???` operator

### DISCUSSION THREAD

**Discussion by Poshi** [johnkerl/miller#1255] [Ideas]
Just run across an operator precedence issue where the results where not as expected (at least, for my mind, as they were as expected according the documentation).
I ran something like:
`put '$x = $a???0 + $b???0'`

The intent was to add two numbers, substituting by zero the empty ones. I was mostly getting the `$a` value, but not the sum of `$a` and `$b`. This makes sense since the precedence of the operators makes that expression equivalent to:
`put '$x = ($a???(0 + $b))???0'`
which is wildly different from what I was expecting.

Would not make sense to increase the operator precedence of ??/??? right after the parenthesis? Unfortunately, that would be a breaking change for old scripts that rely on the current precedence :-(

**Comment by johnkerl** [comment]
@Poshi this is a bug :(.

**Comment by Poshi** [comment]
Then, that bug expanded to the documentation, as that behavior matches the docs :-/
I see you already opened the corresponding bug report. Thanks for acknowledging it.
